impl/src/test/java/slack/SlackClientImplTest.java

```diff
@@ -0,0 +1,178 @@
+        // and the expected response is returned
+        assertThat(response).isEqualTo(usersLookupByEmailResponse);
+    }
+
+
+    @Test
+    public void shouldCallChannelsReplies() throws Exception {
+        // given a user lookup request
+        ChannelsRepliesRequest chatPostMessageRequest = ChannelsRepliesRequest.builder()
+            .token(slackConfig.getApiToken())
+            .build();
+
+        // when we look up user
+        ChannelsRepliesResponse usersLookupByEmailResponse = new ChannelsRepliesResponse();
+        when(methodsClient.channelsReplies(chatPostMessageRequest)).thenReturn(usersLookupByEmailResponse);
+
+        ChannelsRepliesResponse response = slackClient.channelsReplies(chatPostMessageRequest).toBlocking().single();
+        // then the real method should be invoked
+        verify(methodsClient, times(1)).channelsReplies(chatPostMessageRequest);
+        // and the token in the config should be used
+        assertThat(chatPostMessageRequest.getToken()).isEqualTo(slackConfig.getApiToken());
+        // and the expected response is returned
+        assertThat(response).isEqualTo(usersLookupByEmailResponse);
+    }
+
+    @Test
+    public void shouldCallUsersInfo() throws Exception {
+        // given a user lookup request
+        UsersInfoRequest chatPostMessageRequest = UsersInfoRequest.builder()
+            .token(slackConfig.getApiToken())
+            .build();
+
+        // when we look up user
+        UsersInfoResponse usersLookupByEmailResponse = new UsersInfoResponse();
+        when(methodsClient.usersInfo(chatPostMessageRequest)).thenReturn(usersLookupByEmailResponse);
+
+        UsersInfoResponse response = slackClient.usersInfo(chatPostMessageRequest).toBlocking().single();
+        // then the real method should be invoked
+        verify(methodsClient, times(1)).usersInfo(chatPostMessageRequest);
+        // and the token in the config should be used
+        assertThat(chatPostMessageRequest.getToken()).isEqualTo(slackConfig.getApiToken());
+        // and the expected response is returned
+        assertThat(response).isEqualTo(usersLookupByEmailResponse);
+    }
+
+    @Test
+    public void shouldLogOnIOException() throws Exception {
+
+        // given a user lookup request
+        UsersInfoRequest chatPostMessageRequest = UsersInfoRequest.builder()
+            .token(slackConfig.getApiToken())
+            .build();
+
+        // when we look up user on bad network
+        when(methodsClient.usersInfo(chatPostMessageRequest)).thenThrow(new IOException("connection reset by peer"));
+
+        // when we asks slack for the message, then a exception should be returned
+        assertThatExceptionOfType(RuntimeException.class)
+            .isThrownBy(() -> slackClient.usersInfo(chatPostMessageRequest).toBlocking().single())
+            .satisfies(e -> assertThat(e.getMessage()).isEqualTo("java.io.IOException: connection reset by peer"));
```

Comment:
There's a withMessage to be used here. :)

Reply:
ah cool :)